### ISSUE TITLE
🐛 [cloud_firestore] The method '-' was called on null

### ISSUE THREAD

**Issue by chirmac** [FirebaseExtended/flutterfire#3825]
## Bug report

When attempting to get all the documents on a collection (or the QuerySnapshot stream for real time updates), the request fails and returns the error "The method '-' was called on null". The user was properly logged in in firebase_auth.

### Steps to reproduce

1. Authenticate a FirebaseUser
2. Attempt to fetch a document from Firestore

### Expected behavior

No errors when attempting to fetch documents and/or collections.

---

### Flutter doctor

<details><summary>Click To Expand</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel beta, 1.22.0, on Microsoft Windows [Version 10.0.18363.1082], locale pt-PT)

[√] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
[√] Chrome - develop for the web
[!] Android Studio (version 4.0)
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
[√] VS Code (version 1.50.0)
[√] Connected device (3 available)

! Doctor found issues in 1 category.
```

</details>

---

### Flutter dependencies

<details><summary>Click To Expand</summary>

```
Dart SDK 2.10.0
Flutter SDK 1.22.0       
instagram_metrics 1.0.0+1

dependencies:
- authentication_repository 1.0.0 [flutter equatable firebase_auth firebase_core google_sign_in meta]
- bloc 6.0.3 [meta]
- cloud_firestore 0.14.1+3 [flutter meta quiver firebase_core firebase_core_platform_interface cloud_firestore_platform_interface cloud_firestore_web]
- cupertino_icons 1.0.0
- equatable 1.2.5 [collection meta]
- firebase_auth 0.18.1+2 [meta firebase_core firebase_core_platform_interface firebase_auth_platform_interface firebase_auth_web flutter]
- firebase_core 0.5.0+1 [firebase_core_platform_interface flutter quiver meta firebase_core_web]
- flutter 0.0.0 [characters collection meta typed_data vector_math sky_engine]
- flutter_bloc 6.0.6 [flutter bloc provider]
- font_awesome_flutter 8.10.0 [flutter]
- formz 0.2.0
- google_fonts 1.1.0 [flutter http path_provider crypto pedantic]
- meta 1.3.0-nullsafety.3

dev dependencies:
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters charcode collection matcher meta source_span stream_channel string_scanner term_glyph typed_data]

transitive dependencies:
- async 2.5.0-nullsafety.1 [collection]
- boolean_selector 2.1.0-nullsafety.1 [source_span string_scanner]
- characters 1.1.0-nullsafety.3
- charcode 1.2.0-nullsafety.1
- clock 1.1.0-nullsafety.1
- cloud_firestore_platform_interface 2.1.2 [flutter meta collection firebase_core plugin_platform_interface]
- cloud_firestore_web 0.2.0+4 [flutter flutter_web_plugins firebase http_parser meta firebase_core cloud_firestore_platform_interface js]
- collection 1.15.0-nullsafety.3
- convert 2.1.1 [charcode typed_data]
- crypto 2.1.5 [collection convert typed_data]
- fake_async 1.2.0-nullsafety.1 [clock collection]
- ffi 0.1.3
- file 5.2.1 [intl meta path]
- firebase 7.3.0 [http http_parser js]
- firebase_auth_platform_interface 2.1.1 [flutter meta firebase_core plugin_platform_interface]
- firebase_auth_web 0.3.1+1 [flutter flutter_web_plugins firebase meta http_parser intl firebase_core firebase_auth_platform_interface js]
- firebase_core_platform_interface 2.0.0 [flutter meta plugin_platform_interface quiver]
- firebase_core_web 0.2.0 [firebase firebase_core_platform_interface flutter flutter_web_plugins meta js]
- flutter_web_plugins 0.0.0 [flutter characters collection meta typed_data vector_math]
- google_sign_in 4.5.4 [google_sign_in_platform_interface flutter meta google_sign_in_web]
- google_sign_in_platform_interface 1.1.2 [flutter meta quiver]
- google_sign_in_web 0.9.2 [google_sign_in_platform_interface flutter flutter_web_plugins meta js]
- http 0.12.2 [http_parser path pedantic]
- http_parser 3.1.4 [charcode collection source_span string_scanner typed_data]
- intl 0.16.1 [path]
- js 0.6.2
- matcher 0.12.10-nullsafety.1 [stack_trace]
- nested 0.0.4 [flutter]
- path 1.8.0-nullsafety.1
- path_provider 1.6.18 [flutter path_provider_platform_interface path_provider_macos path_provider_linux path_provider_windows]
- path_provider_linux 0.0.1+2 [path xdg_directories path_provider_platform_interface flutter]
- path_provider_macos 0.0.4+4 [flutter]
- path_provider_platform_interface 1.0.3 [flutter meta platform plugin_platform_interface]
- path_provider_windows 0.0.4+1 [path_provider_platform_interface meta path flutter ffi win32]
- pedantic 1.9.2 [meta]
- platform 2.2.1
- plugin_platform_interface 1.0.3 [meta]
- process 3.0.13 [file intl meta path platform]
- provider 4.3.2+2 [flutter nested collection]
- quiver 2.1.3 [matcher meta]
- sky_engine 0.0.99
- source_span 1.8.0-nullsafety.2 [charcode collection path term_glyph]
- stack_trace 1.10.0-nullsafety.1 [path]
- stream_channel 2.1.0-nullsafety.1 [async]
- string_scanner 1.1.0-nullsafety.1 [charcode source_span]
- term_glyph 1.2.0-nullsafety.1
- test_api 0.2.19-nullsafety.2 [async boolean_selector collection meta path source_span stack_trace stream_channel string_scanner term_glyph matcher]
- typed_data 1.3.0-nullsafety.3 [collection]
- vector_math 2.1.0-nullsafety.3
- win32 1.7.3 [ffi]
- xdg_directories 0.1.2 [meta path process]
```

</details>

---

**Comment by Ehesp** [comment]
Could you show a code example please which causes this error?

**Comment by JayPerfetto** [comment]
I also have this issue but only on Android - any query to any document or collection returns a doc with "exists = false" or a query snapshot with 0 documents in it. This is even the case for hardcoded collection and ID names, which I am literally looking at in my database and I can see the queries work on non-android devices.

**Comment by seanpmaddox** [comment]
Am having this issue as well.  @Ehesp if you're still looking for a code sample where this is happening see the snippet below.  Regarding it 1) appUserCollection is set in the class as final appUserCollection = FirebaseFirestore.instance.collection('u'); 2) appUser.dbData.docID is set elsewhere to a string that is known to be a valid document ID in the u collection and 3) just to be sure i was not missing something I hardcoded a reference to another collection/document which you'll see here and confirmed that the issue happens on that one as well.

Am using firebase_auth: ^0.18.0+1 and cloud_firestore: ^0.14.1+3

I've only tried on Android, haven't yet tried on Apple.

  @override
  Stream<AppUser> streamSnapshot(AppUser appUser) {
    print("In FirestoreRepo.streamSnapshot, appUser.dbData.docID = ${appUser
        .dbData.docID}");
//    return appUserCollection.doc(appUser.dbData.docID).snapshots().map((
//        snapshot) {
    return FirebaseFirestore.instance.collection('l').doc('QICB827EB274AD2').snapshots().map((
    snapshot) {
      print(
          "In FirestoreRepo.streamSnapshot w/snapshot fired, appUser.dbData.docID = ${appUser
              .dbData.docID}");
      return AppUser.fromEntity(AppUserEntity.fromSnapshot(appUser, snapshot));
    });
  }
